openedx/core/lib/xblock_utils/__init__.py

```diff
@@ -120,8 +120,10 @@ def wrap_xblock(
             css_classes.append('is-hidden')
 
         css_classes.append('xmodule_' + markupsafe.escape(class_name))
+
+    if isinstance(block, (XModule, XModuleDescriptor)):
         data['type'] = block.js_module_name
-        shim_xmodule_js(block, frag)
+        shim_xmodule_js(frag, block.js_module_name)
```

Comment:
The `ProblemBlock` does load the javascript bundle and shim itself in `student_view()` and `studio_view()`. This wouldn't work because we have merged two classes into one.